unfurl/parsers/parse_youtube.py

```diff
@@ -22,11 +25,32 @@
 }
 
 
+def format_seconds(seconds):
+    if seconds < 60:
+        return f"{seconds} seconds"
+    elif seconds < 3600:
+        return time.strftime("%M:%S", time.gmtime(seconds)) + " minutes"
+    else:
+        return time.strftime("%H:%M:%S", time.gmtime(seconds)) + " hours"
+
+
 def run(unfurl, node):
     youtube_domains = ['youtube.com', 'youtu.be']
     if any(youtube_domain in unfurl.find_preceding_domain(node) for youtube_domain in youtube_domains):
-        if node.key == 't':
-            param_text = f'The youtube video will begin playing at {str(node.value)} seconds.'
+        if node.key == 't' or node.key == 'time_continue' or node.key == 'start':
+            try:
+                time_formatted = format_seconds(int(node.value))
+                param_text = f'The youtube video will begin playing at {time_formatted}.'
+                unfurl.add_to_queue(
+                    data_type='descriptor', key=None, value=node.value, label=param_text,
+                    parent_id=node.node_id, incoming_edge_config=youtube_edge)
+            except:
+                pass
+        if node.key == 'v' or (node.data_type == 'url.path.segment' and (not 'watch' in node.value and not 'embed' in node.value)) or (node.data_type == 'url.path' and len(node.value[1:].split('/')) == 1 and not '/watch' in node.value and not '/embed' in node.value):
```

Comment:
This line is kind of long, but I couldn't figure out an easier way to get the Video ID for urls where the id equals the path and for urls where it equals the path segment.

Tested with:
https://www.youtube.com/watch?time_continue=423&v=5VKTaFBlMcE&feature=emb_logo
https://youtu.be/5VKTaFBlMcE?t=423
https://www.youtube.com/embed/5VKTaFBlMcE

Reply:
From what I can tell, the video ID is always 11 chars long, so maybe checking the length can simplify removing the 'watch' and 'embed' cases? It's also ok that it's long.

Reply:
I agree, checking for length is the better solution.